crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,177 @@
+                return Err(D::Error::custom("SNP migration not supported"));
+            }
+            "SingleSocket" => {
+                flags |= PolicyFlags::SingleSocket;
+            }
+            "SMT" => {
+                flags |= PolicyFlags::SMT;
+            }
+            _ => return Err(D::Error::custom(format!("unknown policy flag '{}'", flag))),
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
+
+fn from_platform_string<'de, D>(deserializer: D) -> Result<Option<PlatformInfoFlags>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::de::Error;
+
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let flag = match s.trim() {
+        "SME" => PlatformInfoFlags::SME,
+        "TSME" => PlatformInfoFlags::TSME,
+        _ => {
+            return Err(D::Error::custom(format!(
+                "unknown platform info flag '{}'",
+                s
+            )))
+        }
+    };
+
+    Ok(Some(flag))
+}
+
+#[derive(Clone, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    #[serde(flatten)]
+    pub config: ConfigIntermediate,
+}
+
+impl<'de> Deserialize<'de> for Config {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        use serde::de::Error;
+
+        let config = ConfigIntermediate::deserialize(deserializer)?;
+
+        if config.enarx_signer.is_empty() && config.enarx_hash.is_empty() {
+            return Err(D::Error::custom(
+                "enarx_signer and enarx_hash cannot both be empty",
+            ));
+        }
+
+        Ok(Self { config })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::config::Config;
+    use crate::PolicyFlags;
+    use flagset::FlagSet;
+
+    #[test]
+    fn test_empty_config() {
+        let result: Result<Config, toml::de::Error> = toml::from_str("");
+        if result.is_ok() {
+            unreachable!();
+        }
+    }
+
+    #[test]
+    fn test_flags() {
+        let config_raw = r#"
+        enarx_signer = ["1234567890", "00112233445566778899"]
+        policy_flags = "SingleSocket | Debug"
+        platform_info_flags = "SME"
+        "#;
+
+        let config_obj: Config = toml::from_str(config_raw).expect("Couldn't deserialize");
+        assert!(config_obj.config.policy_flags.is_some());
+        assert!(config_obj.config.platform_info_flags.is_some());
+
+        let flags = FlagSet::<PolicyFlags>::new(config_obj.config.policy_flags.unwrap()).unwrap();
+        assert_eq!(
+            flags,
+            PolicyFlags::SingleSocket | PolicyFlags::Debug | PolicyFlags::Reserved
+        );
+
+        assert_eq!(config_obj.config.enarx_signer.len(), 2);
+        assert_eq!(
+            config_obj.config.enarx_signer.get(0).unwrap().as_ref(),
+            &hex::decode("1234567890").unwrap()
+        );
+        assert_eq!(
+            config_obj.config.enarx_signer.get(1).unwrap().as_ref(),
+            &hex::decode("00112233445566778899").unwrap()
+        );
```

Comment:
Just call it `config`, the `_obj` is redundant

Reply:
Fixed